exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonEscapeAnyChar.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.vector.complex.writer;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.drill.common.exceptions.UserRemoteException;
+import org.apache.drill.test.BaseTestQuery;
+import org.apache.drill.exec.ExecConstants;
+import org.junit.Test;
+
+import java.io.File;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+
+public class TestJsonEscapeAnyChar extends BaseTestQuery {
+
+  @Test
+  public void testwithOptionEnabled() throws Exception {
+    String table = "escape.json";
+    File file = new File(dirTestWatcher.getRootDir(), table);
+    String json = "{\"name\": \"ABC\\S\"}";
+    String query = String.format("select * from dfs.`%s`", table);
+    try {
+      FileUtils.writeStringToFile(file, json);
+      test("set `%s` = true", ExecConstants.JSON_READER_ESCAPE_ANY_CHAR);
+      testBuilder()
+        .sqlQuery(query)
+        .unOrdered()
+        .baselineColumns("name")
+        .baselineValues("ABCS")
```

Comment:
True. I've escaped the backslash here because i'm constructing the JSON string here in Java. If it's just plain text JSON, the input would be 'ABC\S'. So you can assume the test is actually for the JSON string 'ABC\S'.

Reply:
Makes sense. Thanks for explanation.